### PR TITLE
Add proxy_http_fetch import stub

### PR DESCRIPTION
**Pull request by Thomasvdam**
## Motivation

>[!CAUTION]
> I'm not sure which versions (if any) we need to update and what other steps we need to make sure the chain uses the correct version.

Prevent tally execution from crashing when executing an oracle program that uses the `proxy_http_fetch` import.

## Explanation of Changes

Duplicate of the `http_fetch` import stub. I briefly tried to reduce duplication with a higher order function but the compiler started yelling at me and I wasn't having it.

## Testing

Updated the Rust tests to use a new oracle program binary which imports and uses the `proxy_http_fetch` and `http_fetch` methods.

## Related PRs and Issues

Closes: #18

### CODE REVIEW

**Review by FranklinWaller — APPROVED** [review]

File: libtallyvm/src/lib.rs [hunk 1]
```diff
@@ -207,17 +206,40 @@ mod test {
...
-            vec!["testHttpSuccess".to_string()],
+            vec![hex::encode("testHttpSuccess")],
             HashMap::default(),
         )
         .unwrap();
 
         result.stdout.iter().for_each(|line| print!("{}", line));
 
-        dbg!(result);
+        dbg!(&result);
```
Comment (gluax) [inline libtallyvm/src/lib.rs:219]: `dbg` Statements

**Review by gluax — APPROVED** [review]
We also need to build and push the updated libraries to the repo as a heads up.

File: libtallyvm/src/lib.rs (hunk 1 above)
Comment (gluax) [reply libtallyvm/src/lib.rs:219]: oh lol did we forget to remove that one before

**Comment by Thomasvdam** [comment]
@gluax or @hacheigriega Is there any version number I have to bump or can I just run the build pipeline, download the artifacts, and add them to this PR?

**Comment by gluax** [comment]
> @gluax or @hacheigriega Is there any version number I have to bump or can I just run the build pipeline, download the artifacts, and add them to this PR?

Nope you can do that! The only thing is that the pipeline can't build the Mac version.

**Comment by Thomasvdam** [comment]
I guess it's better if someone who actually knows how the release process works creates the builds in this PR and documents it in the README? I'm not sure which build command I should execute and what of the build output should go where under source control 😬

**Comment by gluax** [comment]
> I guess it's better if someone who actually knows how the release process works creates the builds in this PR and documents it in the README? I'm not sure which build command I should execute and what of the build output should go where under source control 😬

I can update them! I'll update the docs in this PR as well.

**Comment by gluax** [comment]
@Thomasvdam, I built and pushed the ones the CI could make. I also added some documents on how to complete the process.

I can't build the macOS version; could you or @hacheigriega add that one, please?

**Comment by Thomasvdam** [comment]
Thanks Jon ❤️ I can do the macOS build next week! @hacheigriega is there anything special to creating the release once it’s merged?

**Pull request merged**